cloudformation/cw-dashboard-substack.cfn.yaml

```diff
@@ -0,0 +1,554 @@
+            {%- endfor -%}
+            {{- update_coord(graph_width, graph_height) }}
+          {%- endfor -%}
+          {{- update_coord_after_section(graph_height) }}
+
+          {#- EBS metrics graphs #}
+          - ',{"type":"text","x":{{ coord.x }},"y":{{ coord.y }},"width":24,"height":1,"properties":{"markdown":"\n##
+            EBS Metrics\n"}}'
+          {{- update_coord_after_section(1) }}
+
+          {#- Unconditional EBS metrics #}
+          {%- set ebs_metrics = [{'metrics': ["VolumeReadOps", "VolumeWriteOps"], 'extra_params': ['"title":"Read/Write Ops"']},
+                                 {'metrics': ["VolumeReadBytes", "VolumeWriteBytes"], 'extra_params': ['"title":"Read/Write Bytes"']},
+                                 {'metrics': ["VolumeTotalReadTime", "VolumeTotalWriteTime"], 'extra_params': ['"title":"Total Read/Write Time"']},
+                                 {'metrics': ["VolumeQueueLength"], 'extra_params': ['"title":"Queue Length"']},
+                                 {'metrics': ["VolumeIdleTime"], 'extra_params': ['"title":"Idle Time"']}]
+          -%}
+          {%- set number_of_ebs_volumes = cfn_params.NumberOfEBSVol|int -%}
+          {%- set ebs_volume_types = cfn_params.VolumeType.split(',') %}
+
+          {%- for metrics_param in ebs_metrics %}
+          - !Sub ',{"type":"metric","x":{{ coord.x }},"y":{{ coord.y }},"width":{{ graph_width }},"height":{{ graph_height }},"properties":{"view":"timeSeries","stacked":false,
+            {{ metrics_param.extra_params|join(',') }}{% if metrics_param.extra_params|length > 0 %},{% endif %}"region":"${AWS::Region}","metrics":'
+            {%- for i in range(number_of_ebs_volumes) %}
+              {%- set volume_loop = loop %}
+              {%- for metric in metrics_param.metrics %}
+          - !Sub
+            - '{% if volume_loop.first and loop.first %}[{% else %},{% endif %}["AWS/EBS","{{ metric }}","VolumeId","${EBS_Volume{{ i|int +1 }}}"]{% if volume_loop.last and loop.last %}]}}{% endif %}'
+            - EBS_Volume{{ i|int +1 }}: !Select
+                - '{{ i|int }}'
+                - !Ref 'EBSVolumesIds'
+              {%- endfor %}
+            {%- endfor %}
+          {{- update_coord(graph_width, graph_height) -}}
+          {%- endfor %}
+
+          {#- Conditional EBS metrics #}
+          {%- set ebs_metrics_conditions = [{'metric': 'VolumeConsumedReadWriteOps', 'supported_vol_types': ["io1"], 'extra_params': ['"title":"Consumed Read/Write Ops"']},
+                                            {'metric': 'VolumeThroughputPercentage', 'supported_vol_types': ["io1"], 'extra_params': ['"title":"Throughput Percentage"']},
+                                            {'metric': 'BurstBalance', 'supported_vol_types': ["gp2", "st1", "sc1"], 'extra_params': ['"title":"Burst Balance"']}]
+          %}
+          {%- for metric_condition_params in ebs_metrics_conditions %}
```

Comment:
It seems to me we could unify the conditional/unconditional loops

Reply:
Let's talk about this, we could do it but would it be very clear?